internal/clientconn/conninfo/conn_info.go

```diff
@@ -53,6 +58,29 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// Cursor returns the cursor value stored.
+func (connInfo *ConnInfo) Cursor(_ uint64) iterator.Interface[int, any] {
+	connInfo.curRW.RLock()
+	defer connInfo.curRW.RUnlock()
+
+	return connInfo.cursor
+}
+
+// SetCursor stores the cursor value.
+// We use "db.collection" as the key to store the cursor.
+func (connInfo *ConnInfo) SetCursor(iterator iterator.Interface[int, any]) {
+	connInfo.curRW.Lock()
+	defer connInfo.curRW.Unlock()
+
+	if connInfo.cursor != nil {
+		connInfo.cursor.Close()
+
+		connInfo.cursor = nil
```

Comment:
That line does nothing, as `connInfo.cursor` is overwritten immediately after that

Reply:
Removed that line.